File: init_database.py
#!/usr/bin/env python3
"""Simple script to initialize the merge-god database"""

import sys
from pathlib import Path

# Add project to path
sys.path.insert(0, str(Path(__file__).parent))

from merge_god.db_operations import StateDatabase  # noqa: E402


def main():
    """Initialize database with schema"""
    db_path = Path("merge-god-state.db")

    print(f"Creating database at: {db_path}")
    db = StateDatabase(db_path)
    print("✅ Database initialized successfully!")

    # Get statistics
    stats = db.get_statistics()
    print("\nDatabase statistics:")
    for key, value in stats.items():
        print(f"  {key}: {value}")

    return 0


if __name__ == "__main__":
    sys.exit(main())

File: init_database.ts
#!/usr/bin/env node
/**
 * Initialize the merge-god database (both the @merge-god/github-sync SyncStore
 * tables and the merge-god AppStore agent/session tables).
 */

import { pathToFileURL } from "node:url";
import { SyncStore } from "@merge-god/github-sync";
import { AppStore } from "./app_store";

async function main(): Promise<number> {
  const dbPath = "merge-god-state.db";

  console.log(`Creating database at: ${dbPath}`);
  const syncStore = new SyncStore(dbPath);
  await syncStore.initialize();
  const appStore = new AppStore(dbPath);
  console.log("✅ Database initialized successfully!");

  const stats = (await syncStore.getStatistics()) as Record<string, unknown>;
  console.log("\nDatabase statistics:");
  for (const [key, value] of Object.entries(stats)) {
    console.log(`  ${key}: ${value}`);
  }

  syncStore.close();
  appStore.close();
  return 0;
}

if (import.meta.url === pathToFileURL(process.argv[1] ?? "").href) {
  void main().then((code) => process.exit(code));
}
